server/src/test/java/org/elasticsearch/index/mapper/MappingParserTests.java

```diff
@@ -207,132 +212,151 @@ to get the wrong path (missing the first portion).
         assertEquals("obj.source.geo.location", geoPointFieldMapper.mappedFieldType.name());
     }
 
+    private static String randomFieldType() {
+        return randomBoolean() ? KeywordFieldMapper.CONTENT_TYPE : ObjectMapper.CONTENT_TYPE;
```

Comment:
good idea

Reply:
I tried it but it won't work easily because field alias need to have path declared. I feel like this is ok, we care to see that the checks are not made for a specific type only, and we still test for e.g. empty name in MapperTestCase.